internal/fakecluster/fakecluster.go

```diff
@@ -7,34 +7,26 @@ import (
 	"strings"
 
 	"github.com/ethereum/go-ethereum/p2p/discv5"
-	"github.com/vipnode/vipnode/client"
-	"github.com/vipnode/vipnode/host"
+	"github.com/vipnode/vipnode/agent"
 	"github.com/vipnode/vipnode/internal/fakenode"
 	"github.com/vipnode/vipnode/jsonrpc2"
 	"github.com/vipnode/vipnode/pool"
 	"github.com/vipnode/vipnode/pool/store/memory"
 )
 
-type clusterHost struct {
-	*host.Host
-	Node *fakenode.FakeNode
-	In   *jsonrpc2.Remote
-	Out  *jsonrpc2.Remote
-	Key  *ecdsa.PrivateKey
-}
-
-type clusterClient struct {
-	*client.Client
-	Node *fakenode.FakeNode
-	In   *jsonrpc2.Remote
-	Out  *jsonrpc2.Remote
-	Key  *ecdsa.PrivateKey
+type clusterAgent struct {
+	*agent.Agent
+	Node       *fakenode.FakeNode
+	RemotePool pool.Pool
+	In         *jsonrpc2.Remote
+	Out        *jsonrpc2.Remote
+	Key        *ecdsa.PrivateKey
 }
 
 // Cluster represents a set of active hosts and clients connected to a pool.
 type Cluster struct {
```

Comment:
fakecluster is still a bit more Host/Client-specific than I'd like. Not super clear whether that distinction is even useful for testing too much, or should just be a side effect of something else. The